apollo-router/src/plugins/include_subgraph_errors/mod.rs

```diff
@@ -0,0 +1,169 @@
+                    return Err(format!(
+                        "Subgraph '{}' must use boolean config when global config is boolean",
+                        name
+                    )
+                    .into());
+                }
+            }
+        }
+
+        // Generate and store the effective configuration
+        let config = Arc::new(init.config.try_into()?);
+
+        Ok(IncludeSubgraphErrors { config })
+    }
+
+    fn supergraph_service(&self, service: BoxService) -> BoxService {
+        let config = Arc::clone(&self.config);
+
+        service
+            .map_response(move |response: SupergraphResponse| {
+                response.map_stream(move |mut graphql_response: graphql::Response| {
+                    for error in &mut graphql_response.errors {
+                        Self::process_error(&config, error);
+                    }
+                    for incremental in &mut graphql_response.incremental {
+                        for error in &mut incremental.errors {
+                            Self::process_error(&config, error);
+                        }
+                    }
+
+                    graphql_response
+                })
+            })
+            .boxed()
+    }
+
+    fn subgraph_service(
+        &self,
+        subgraph_name: &str,
+        service: crate::services::subgraph::BoxService,
+    ) -> crate::services::subgraph::BoxService {
+        // We need to attach the subgraph name to each error so that we can do the filtering in the supergraph service.
+        // The reason filtering is not done here is that other types of request may also generate errors that need filtering.
+        // Pushing the error filtering to supergraph will ensure that everything gets filtered.
+        let subgraph_name = subgraph_name.to_string();
+        service
+            .map_response(move |mut r| {
+                let body = r.response.body_mut();
+                for error in &mut body.errors {
+                    error.add_subgraph_name(&subgraph_name);
+                }
+                r
+            })
+            .boxed()
+    }
+}
+
+impl IncludeSubgraphErrors {
+    fn process_error(config: &Arc<EffectiveConfig>, error: &mut Error) {
+        if let Some(subgraph_name) = error.subgraph_name() {
+            // Get the effective config for this specific subgraph, or use default
+            let effective_config = config
+                .subgraphs
+                .get(&subgraph_name)
+                .unwrap_or(&config.default);
+
+            if !effective_config.include_errors {
+                tracing::debug!(
+                    "Redacting errors for subgraph '{}' based on config: include_errors=false",
+                    subgraph_name
+                );
+                // Redact fully if errors should not be included
+                error.message = REDACTED_ERROR_MESSAGE.to_string();
+                error.extensions = Object::new(); // Clear all extensions
```

Comment:
Do we make it clear in the docs that redaction of errors includes all error extensions?

Reply:
[e549e03](https://github.com/apollographql/router/pull/7164/commits/e549e038c362c07786f35c6739b7c96446571e1b)